src/cli/one_helper/oneprovision_helper.rb

```diff
@@ -14,652 +14,593 @@
+    def ask_deployment(deployment_confs)
+        return if deployment_confs.nil? || deployment_confs.empty?
 
-            return OpenNebula::Error.new(rc[1]) unless rc[0] == 0
+        puts 'Please select a deployment configuration for this provision:'
 
-            rc[1]
+        deployment_confs.each_with_index do |conf, index|
+            name = conf[:name]
+            puts "    #{index}: #{name}"
         end
-    end
+        puts
 
-    # Returns a new provision object
-    #
-    # @param id [Integer] Provision ID
-    def factory(id = nil)
-        if id
-            OneProvision::Provision.new_with_id(id, @client)
-        else
-            OneProvision::Provision.new(OneProvision::Provision.build_xml,
-                                        @client)
-        end
-    end
+        selected = nil
 
-    # Returns provision pool
-    def factory_pool(_)
-        OneProvision::ProvisionPool.new(@client)
-    end
+        loop do
+            print '    Please type the selection number: '
+            input = STDIN.readline.strip
 
-    # Returns helper and filter depending on type
-    #
-    # @param type [String] Helper type
-    #
-    # @returns [OpenNebula::Helper] Helper
-    def helper(type)
-        case type
-        when 'HOSTS'         then helper = OneHostHelper.new
-        when 'DATASTORES'    then helper = OneDatastoreHelper.new
-        when 'NETWORKS'      then helper = OneVNetHelper.new
-        when 'CLUSTERS'      then helper = OneClusterHelper.new
-        when 'IMAGES'        then helper = OneImageHelper.new
-        when 'TEMPLATES'     then helper = OneTemplateHelper.new
-        when 'VNTEMPLATES'   then helper = OneVNTemplateHelper.new
-        when 'FLOWTEMPLATES' then helper = OneFlowTemplateHelper.new
+            if input =~ /\A\d+\z/
+                index = input.to_i
+
+                if index >= 0 && index < deployment_confs.size
+                    selected = deployment_confs[index]
+                    break
+                end
+            end
+
+            puts '    Invalid selection, please try again.'
         end
 
-        helper.set_client({})
-        helper
+        puts
+        selected
     end
 
-    # Shows resources
-    #
-    # @param resources_names [Array]  Resources types
-    # @param resources       [Array]  Resources information
-    # @param type            [String] Resource or Infrastructure resources
-    def show_resources(resources_names, resources, type)
-        puts
-        CLIHelper.print_header("Provision #{type} Resources")
+    def ask_user_inputs(inputs)
+        puts 'There are some parameters that require user input.'
+
+        answers = {}
+
+        inputs.each do |input|
+            name        = input[:name]
+            description = input[:description] || ''
+            type        = input[:type]
+            default     = input[:default]
+            match       = input[:match]
+
+            puts "  * (#{name}) #{description} [type: #{type}]"
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

A cross-site scripting (XSS) vulnerability was detected. User-controlled data is being output without proper sanitization, allowing an attacker to inject malicious scripts. This could lead to session hijacking, malware installation, or phishing attacks. Always sanitize and encode user inputs before including them in the output, using context-appropriate encoding methods. Learn More - https://cwe.mitre.org/data/definitions/79.html.